test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/PreserveTypesAndAttributesTests.cs

```diff
@@ -0,0 +1,395 @@
e Str. 57\",\"ShipCity\":\"Berlin\",\"ShipRegion\":null,\"ShipPostalCode\":\"12209\",\"ShipCountry\":\"Germany\"},{\"OrderID\":10952,\"CustomerID\":\"ALFKI\",\"EmployeeID\":1,\"OrderDate\":\"1998-03-16T00:00:00Z\",\"RequiredDate\":\"1998-04-27T00:00:00Z\",\"ShippedDate\":\"1998-03-24T00:00:00Z\",\"ShipVia\":1,\"Freight\":40.4200,\"ShipName\":\"Alfred's Futterkiste\",\"ShipAddress\":\"Obere Str. 57\",\"ShipCity\":\"Berlin\",\"ShipRegion\":null,\"ShipPostalCode\":\"12209\",\"ShipCountry\":\"Germany\"},{\"OrderID\":11011,\"CustomerID\":\"ALFKI\",\"EmployeeID\":3,\"OrderDate\":\"1998-04-09T00:00:00Z\",\"RequiredDate\":\"1998-05-07T00:00:00Z\",\"ShippedDate\":\"1998-04-13T00:00:00Z\",\"ShipVia\":1,\"Freight\":1.2100,\"ShipName\":\"Alfred's Futterkiste\",\"ShipAddress\":\"Obere Str. 57\",\"ShipCity\":\"Berlin\",\"ShipRegion\":null,\"ShipPostalCode\":\"12209\",\"ShipCountry\":\"Germany\"}]},{\"CustomerID\":\"ANATR\",\"CompanyName\":\"Ana Trujillo Emparedados y helados\",\"ContactName\":\"Ana Trujillo\",\"ContactTitle\":\"Owner\",\"Address\":\"Avda. de la Constituci\\u00f3n 2222\",\"City\":\"M\\u00e9xico D.F.\",\"Region\":null,\"PostalCode\":\"05021\",\"Country\":\"Mexico\",\"Phone\":\"(5) 555-4729\",\"Fax\":\"(5) 555-3745\",\"Orders\":[{\"OrderID\":10308,\"CustomerID\":\"ANATR\",\"EmployeeID\":7,\"OrderDate\":\"1996-09-18T00:00:00Z\",\"RequiredDate\":\"1996-10-16T00:00:00Z\",\"ShippedDate\":\"1996-09-24T00:00:00Z\",\"ShipVia\":3,\"Freight\":1.6100,\"ShipName\":\"Ana Trujillo Emparedados y helados\",\"ShipAddress\":\"Avda. de la Constituci\\u00f3n 2222\",\"ShipCity\":\"M\\u00e9xico D.F.\",\"ShipRegion\":null,\"ShipPostalCode\":\"05021\",\"ShipCountry\":\"Mexico\"},{\"OrderID\":10625,\"CustomerID\":\"ANATR\",\"EmployeeID\":3,\"OrderDate\":\"1997-08-08T00:00:00Z\",\"RequiredDate\":\"1997-09-05T00:00:00Z\",\"ShippedDate\":\"1997-08-14T00:00:00Z\",\"ShipVia\":1,\"Freight\":43.9000,\"ShipName\":\"Ana Trujillo Emparedados y helados\",\"ShipAddress\":\"Avda. de la Constituci\\u00f3n 2222\",\"ShipCity\":\"M\\u00e9xico D.F.\",\"ShipRegion\":null,\"ShipPostalCode\":\"05021\",\"ShipCountry\":\"Mexico\"},{\"OrderID\":10759,\"CustomerID\":\"ANATR\",\"EmployeeID\":3,\"OrderDate\":\"1997-11-28T00:00:00Z\",\"RequiredDate\":\"1997-12-26T00:00:00Z\",\"ShippedDate\":\"1997-12-12T00:00:00Z\",\"ShipVia\":3,\"Freight\":11.9900,\"ShipName\":\"Ana Trujillo Emparedados y helados\",\"ShipAddress\":\"Avda. de la Constituci\\u00f3n 2222\",\"ShipCity\":\"M\\u00e9xico D.F.\",\"ShipRegion\":null,\"ShipPostalCode\":\"05021\",\"ShipCountry\":\"Mexico\"},{\"OrderID\":10926,\"CustomerID\":\"ANATR\",\"EmployeeID\":4,\"OrderDate\":\"1998-03-04T00:00:00Z\",\"RequiredDate\":\"1998-04-01T00:00:00Z\",\"ShippedDate\":\"1998-03-11T00:00:00Z\",\"ShipVia\":3,\"Freight\":39.9200,\"ShipName\":\"Ana Trujillo Emparedados y helados\",\"ShipAddress\":\"Avda. de la Constituci\\u00f3n 2222\",\"ShipCity\":\"M\\u00e9xico D.F.\",\"ShipRegion\":null,\"ShipPostalCode\":\"05021\",\"ShipCountry\":\"Mexico\"}]}]";
```

Comment:
You don't need such a big payload to verify that the code is working as expected. 2 or at most 3 objects in the JSON array are sufficient...